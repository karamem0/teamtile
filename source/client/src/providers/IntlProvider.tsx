//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import {
  IntlProvider as Provider,
  createIntl,
  MessageFormatElement
} from 'react-intl';

import en from '../translations/compiled/en.json';
import ja from '../translations/compiled/ja.json';

const translations: { [key: string]: Record<string, MessageFormatElement[]> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language.substring(0, 2)]
});

function IntlProvider(props: React.PropsWithChildren<unknown>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
