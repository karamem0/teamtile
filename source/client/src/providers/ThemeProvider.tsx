//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { Provider, themeNames } from '@fluentui/react-teams';

import { app } from '@microsoft/teams-js';

import { inTeams } from '../utils/Teams';

interface ThemeProviderProps {
  children?: React.ReactNode
}

function ThemeProvider(props: ThemeProviderProps) {

  const { children } = props;

  const [ lang, setLang ] = React.useState<string>('en-US');
  const [ themeName, setThemeName ] = React.useState<themeNames>(themeNames.Default);

  const handleThemeChange = (value: string) => {
    switch (value) {
      case 'dark':
        setThemeName(themeNames.Dark);
        break;
      case 'contrast':
        setThemeName(themeNames.HighContrast);
        break;
      default:
        setThemeName(themeNames.Default);
        break;
    }
  };

  React.useEffect(() => {
    if (!inTeams()) {
      return;
    }
    (async () => {
      await app.initialize();
      const context = await app.getContext();
      setLang(context.app.locale);
      handleThemeChange(context.app.theme);
      app.registerOnThemeChangeHandler(handleThemeChange);
    })();
  }, []);

  return (
    <Provider
      lang={lang}
      themeName={themeName}>
      {children}
    </Provider>
  );

}

export default ThemeProvider;
