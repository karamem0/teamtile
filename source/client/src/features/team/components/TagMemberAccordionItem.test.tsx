//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { Accordion } from '@fluentui/react-components';
import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './TagMemberAccordionItem.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';
import { render } from '@testing-library/react';

it('should create a shapshot when the loading parameter is true', () => {
  const params = {
    loading: true,
    members: [
      {
        id: 'MjQzMmI1N2ItMGFiZC00M2RiLWFhN2ItMTZlYWRkMTE1ZDM0IyNlYjY1M2Y5Mi04MzczLTRkZTYtYmZlYy01YjRkMjE2YjZhZGUjI2QzYjJiM2ViLWM0N2YtNDViOS05NWYyLWIyZjJlZjYyMTVjZQ==',
        displayName: 'Adele Vance',
        tenantId: 'dcd219dd-bc68-4b9b-bf0b-4a33a796be35',
        userId: '87d349ed-44d7-43e1-9a83-5f2406dee5bd'
      }
    ],
    tag: {
      id: 'MjQzMmI1N2ItMGFiZC00M2RiLWFhN2ItMTZlYWRkMTE1ZDM0IyM3ZDg4M2Q4Yi1hMTc5LTRkZDctOTNiMy1hOGQzZGUxYTIxMmUjI3RhY29VSjN2RGk==',
      displayName: 'Finance',
      description: 'Finance Team for Mach 8 Project',
      memberCount: 2
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Accordion
          openItems={[
            params.tag
          ]}>
          <Presenter {...params} />
        </Accordion>
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should create a shapshot when the loading parameter is false', () => {
  const params = {
    loading: false,
    members: [
      {
        id: 'MjQzMmI1N2ItMGFiZC00M2RiLWFhN2ItMTZlYWRkMTE1ZDM0IyNlYjY1M2Y5Mi04MzczLTRkZTYtYmZlYy01YjRkMjE2YjZhZGUjI2QzYjJiM2ViLWM0N2YtNDViOS05NWYyLWIyZjJlZjYyMTVjZQ==',
        displayName: 'Adele Vance',
        tenantId: 'dcd219dd-bc68-4b9b-bf0b-4a33a796be35',
        userId: '87d349ed-44d7-43e1-9a83-5f2406dee5bd'
      }
    ],
    tag: {
      id: 'MjQzMmI1N2ItMGFiZC00M2RiLWFhN2ItMTZlYWRkMTE1ZDM0IyM3ZDg4M2Q4Yi1hMTc5LTRkZDctOTNiMy1hOGQzZGUxYTIxMmUjI3RhY29VSjN2RGk==',
      displayName: 'Finance',
      description: 'Finance Team for Mach 8 Project',
      memberCount: 2
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Accordion
          openItems={[
            params.tag
          ]}>
          <Presenter {...params} />
        </Accordion>
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
