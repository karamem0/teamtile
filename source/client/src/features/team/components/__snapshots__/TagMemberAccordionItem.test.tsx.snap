// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot when the loading is false 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

.emotion-1>button {
  padding: 0;
}

.emotion-2 {
  margin: 0;
}

.emotion-3 {
  display: grid;
  grid-template-columns: auto auto;
  grid-gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -ms-flex-pack: left;
  -webkit-justify-content: left;
  justify-content: left;
  padding: 0.5rem;
}

.emotion-3:hover {
  background-color: #f5f5f5;
}

<div
    class="fui-FluentProvider fui-FluentProviderr2 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    >
      <div
        class="fui-Accordion"
      >
        <div
          class="fui-AccordionItem"
        >
          <div
            class="fui-AccordionHeader emotion-1 ___kex8dp0_1udlp87 f19n0e5 f1c21dwh f1s184ao ft85np5"
          >
            <button
              aria-disabled="true"
              aria-expanded="true"
              class="fui-AccordionHeader__button ___ecb1fz0_z658t20 f1ewtqcl f1u2r49w f1ym3bx4 fk6fouc fkhj508 f1i3iumi f1gl81tg f1rmphuq f37px4s fgusgyc f10pi13n fly5x3f fre7gi1 f1358rze fqdk4by f1rvrf73 f5pgtk9 f22iagw f122n59 f1k6fduh figsok6 ftqa4ok f2hkw1w f8hki3x f1d2448m f1bjia2o ffh67wi f226i61 f13kzufm flujwa2 fsx75g8 f15bsgw9 f14e48fq f18yb2kv fd6o370 ffwy5si f3znvyf f57olzd f4stah7 f480a47 fs1por5"
              type="button"
            >
              <span
                aria-hidden="true"
                class="fui-AccordionHeader__expandIcon ___7oynyr0_1pz5o7c f1l02sjl f22iagw f122n59 f106mvju f1pp30po f1vdfbxk"
              >
                <svg
                  aria-hidden="true"
                  class="___12fm75w_1fmvgpr f1w7gpdv fez10in fg4l7m0"
                  fill="currentColor"
                  height="1em"
                  style="transform: rotate(90deg); transition: transform 200ms ease-out;"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.65 4.15c.2-.2.5-.2.7 0l5.49 5.46c.21.22.21.57 0 .78l-5.49 5.46a.5.5 0 0 1-.7-.7L12.8 10 7.65 4.85a.5.5 0 0 1 0-.7Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              Finance
            </button>
          </div>
          <div
            class="fui-AccordionPanel emotion-2 ___1rufncu_1hx1scr f1axvtxu"
          >
            <span
              class="fui-Text emotion-3 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
            >
              <span
                aria-label="Adele Vance"
                class="fui-Avatar r81b29z ___3oygws0_1lfz49j fq4mcun frvgh55 f13mqy1h"
                id="avatar-r3"
                role="img"
              >
                <span
                  class="fui-Avatar__initials rip04v ___105uvta_5ja1qh0 f1f3ti53 fu4yj0j"
                  id="avatar-r3__initials"
                >
                  AV
                </span>
              </span>
              <span
                class="fui-Text ___gbswv60_1h9bfvt fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f1cmbuwj"
              >
                Adele Vance
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the loading is true 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

.emotion-1>button {
  padding: 0;
}

.emotion-2 {
  margin: 0;
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    >
      <div
        class="fui-Accordion"
      >
        <div
          class="fui-AccordionItem"
        >
          <div
            class="fui-AccordionHeader emotion-1 ___kex8dp0_1udlp87 f19n0e5 f1c21dwh f1s184ao ft85np5"
          >
            <button
              aria-disabled="true"
              aria-expanded="true"
              class="fui-AccordionHeader__button ___ecb1fz0_z658t20 f1ewtqcl f1u2r49w f1ym3bx4 fk6fouc fkhj508 f1i3iumi f1gl81tg f1rmphuq f37px4s fgusgyc f10pi13n fly5x3f fre7gi1 f1358rze fqdk4by f1rvrf73 f5pgtk9 f22iagw f122n59 f1k6fduh figsok6 ftqa4ok f2hkw1w f8hki3x f1d2448m f1bjia2o ffh67wi f226i61 f13kzufm flujwa2 fsx75g8 f15bsgw9 f14e48fq f18yb2kv fd6o370 ffwy5si f3znvyf f57olzd f4stah7 f480a47 fs1por5"
              type="button"
            >
              <span
                aria-hidden="true"
                class="fui-AccordionHeader__expandIcon ___7oynyr0_1pz5o7c f1l02sjl f22iagw f122n59 f106mvju f1pp30po f1vdfbxk"
              >
                <svg
                  aria-hidden="true"
                  class="___12fm75w_1fmvgpr f1w7gpdv fez10in fg4l7m0"
                  fill="currentColor"
                  height="1em"
                  style="transform: rotate(90deg); transition: transform 200ms ease-out;"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.65 4.15c.2-.2.5-.2.7 0l5.49 5.46c.21.22.21.57 0 .78l-5.49 5.46a.5.5 0 0 1-.7-.7L12.8 10 7.65 4.85a.5.5 0 0 1 0-.7Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              Finance
            </button>
          </div>
          <div
            class="fui-AccordionPanel emotion-2 ___1rufncu_1hx1scr f1axvtxu"
          >
            <div
              class="fui-Spinner r82apo5"
              role="progressbar"
            >
              <span
                class="fui-Spinner__spinner rvgcg50 ___1xxu43e_eahmz60 f1d2rq10 f1szoe96 fb52u90"
              >
                <span
                  class="fui-Spinner__spinnerTail rxov3xa"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
